plugins/env/app/controllers/environment_variable_groups_controller.rb

```diff
@@ -42,13 +42,14 @@ def destroy
   end
 
   def preview
+    @deploy_groups = !params[:scope].empty? ? DeployGroup.where(permalink: params[:scope]) : DeployGroup.all
     @groups =
       if params[:group_id]
         @group = EnvironmentVariableGroup.find(params[:group_id])
-        SamsonEnv.env_groups(Project.new(environment_variable_groups: [@group]), DeployGroup.all, preview: true)
+        SamsonEnv.env_groups(Project.new(environment_variable_groups: [@group]), @deploy_groups, preview: true)
       else
         @project = Project.find(params[:project_id])
-        SamsonEnv.env_groups(@project, DeployGroup.all, preview: true)
+        SamsonEnv.env_groups(@project, @deploy_groups, preview: true)
```

Comment:
also kinda copy-pasty here ... would be nice to extract project to a standalone if/else and then just have 1 `SamsonEnv.env_groups(@project, @deploy_groups, preview: true)`